modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/views/table/TableCell.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayTable from '@clayui/table';
+import React, {useContext, useEffect, useState} from 'react';
+
+import DataSetDisplayContext from '../../DataSetDisplayContext';
+import CommentRenderer from '../../data_renderers/CommentRenderer';
+import DefaultRenderer from '../../data_renderers/DefaultRenderer';
+import InputCheckboxRenderer from '../../data_renderers/InputCheckboxRenderer';
+import InputDateRenderer from '../../data_renderers/InputDateTimeRenderer';
+import InputTextRenderer from '../../data_renderers/InputTextRenderer';
+import {
+	getDataRendererById,
+	getDataRendererByURL,
+} from '../../utilities/dataRenderers';
+
+function InlineEditInputRenderer({type, value, ...otherProps}) {
+	const {modifiedItems, updateItem} = useContext(DataSetDisplayContext);
+	let InputRenderer;
+
+	switch (type) {
```

Comment:
This could be done inside a `useEffect` so it's only calculated if `type` changes